runtimes/js/src/request_meta.rs

```diff
@@ -32,6 +55,7 @@ pub fn meta(req: &model::Request) -> Result<RequestMeta, serde_json::Error> {
                     .map(serde_json::to_value)
                     .transpose()?,
                 headers: serialize_headers(&rpc.req_headers),
+                middleware_data: req.middleware_data.lock()?.clone().map(PVals),
```

Comment:
I think this should be fine, as there shouldn't be any contention on this lock, but alternatively we can add a function that fetches this data on demand instead